celest/units/quantity.py

```diff
@@ -127,6 +131,54 @@ def __rtruediv__(self, other):
         else:
             return NotImplemented
 
+    def __eq__(self, other):
```

Comment:
There is a lot of repeated code in the comparison dunder methods. Look into refactoring to remove repeated code.